src/miner.cpp

```diff
@@ -544,6 +541,11 @@ bool BlockAssembler::AttemptToAddContractToBlock(CTxMemPool::txiter iter){
 
     ByteCodeExecResult testExecResult = exec.processingResults();
 
+    if(bceResult.usedGas + testExecResult.usedGas > blockGasLimit){
```

Comment:
This is the wrong time to reject the transaction. It can be a DoS attack vector. The condition should be before performByteCode and could be something like `if(txGasLimit > remainingBlockGas) return false`